flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceContinuous.java

```diff
@@ -367,6 +382,8 @@ public void testSpecificSnapshotTimestamp() throws Exception {
 
       List<Row> result3 = waitForResult(iter, 2);
       TestHelpers.assertRecords(result3, batch3, tableResource.table().schema());
+
+      assertThatIcebergEnumeratorMetricsExist();
```

Comment:
Do we want tests for asserting the metrics values too?

Reply:
I guess it is probably difficult to reliably assert on the values of  `unassginedSplits` and `pendingRecords` due to timing, unless we can add a listener to the metric reporter to track all value changes for a metric.

Reply:
We might be able to wait for the expected metrics values, like we do it here:
https://github.com/apache/iceberg/blob/13d2160bd00e08e1e700a980f2a3e64017211bb8/flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java#L338

Reply:
polling won't work because the value may have changed from 0 to 1 and back to 0 within a polling interval.

Reply:
Yeah polling is a challenge. I can do fine-grained units to have better control on when it is invoked. However, I don't think it is possible to assert on a distinct value

Reply:
Makes sense. Thanks for the explanation!